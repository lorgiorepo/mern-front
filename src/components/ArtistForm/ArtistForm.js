import React from 'react';
import {
    Box,
    Button,
    Card,
    CardContent,
    CardHeader,
    Divider,
    Grid,
    TextField,
    makeStyles
  } from '@material-ui/core';

const ArtistForm = () => {
    return (
    <form>
    <Card>
        <CardHeader
          subheader="The information can be edited"
          title="Artist Profile"
        />
        <Divider />
        <CardContent>
            <Grid container spacing={3}>
                <Grid item md={6} xs={12}>
                    <TextField fullWidth
                        helperText="Please specify the first name"
                        label="First name"
                        name="firstName"
                        required
                        variant="outlined"/>
                </Grid>
                <Grid item md={6} xs={12}>
                    <TextField
                        fullWidth
                        label="Last name"
                        name="lastName"
                        required
                        variant="outlined"
                    />
                </Grid>
                <Grid item md={6} xs={12}>
                    <TextField
                        fullWidth
                        label="Nationality"
                        name="nationality"
                        required
                        variant="outlined"
                    />
                </Grid>
                <Grid item md={6} xs={12}>
                    <TextField
                        fullWidth
                        label="About"
                        name="about"
                        required
                        variant="outlined"
                    />
                </Grid>
            </Grid>
        </CardContent>
        <Divider />
        <Box display="flex" justifyContent="flex-end" p={2}>
            <Button color="primary" variant="contained">
                Create
            </Button>
        </Box>
    </Card>
    </form>
    );
}
export default ArtistForm;
