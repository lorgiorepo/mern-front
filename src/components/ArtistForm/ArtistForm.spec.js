import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import promise from 'redux-promise-middleware';
import configureStore from 'redux-mock-store';
import ArtistForm from './ArtistForm';

describe('components > ArtistForm', () => {

    const mockStore = configureStore([promise]);
    const mockAxios = new MockAdapter(axios);

    beforeEach(() => {
        mockAxios.resetHandlers();
    });

    it('renders artist form', () => {
        const store = mockStore({
            artistsData: {
                artists: []
            },
        });

        const {asFragment} = render(<ArtistForm />, {
            wrapper: ({children}) => <Provider store={store}>{children}</Provider>,
        });

        expect(asFragment()).toMatchSnapshot();
    });
});