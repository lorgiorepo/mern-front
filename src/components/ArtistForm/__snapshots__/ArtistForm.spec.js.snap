// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > ArtistForm renders artist form 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Create Artist
    </h1>
    <form>
      <input
        placeholder="Enter Artist Name"
        required=""
        type="text"
      />
      <br />
      <br />
      <input
        placeholder="Enter Artist Lastname"
        required=""
        type="text"
      />
      <br />
      <br />
      <input
        placeholder="Nationality"
        required=""
        type="text"
      />
      <br />
      <br />
      <textarea
        cols="28"
        placeholder="Enter Biography"
        required=""
        rows="5"
      />
      <br />
      <br />
      <button
        type="button"
      >
        Create
      </button>
    </form>
  </div>
</DocumentFragment>
`;
